### PR TITLE
feat: [362] connect intent to service

### PR DESCRIPTION
**Pull request by bartekpospiech**
related tasks

https://github.com/buerokratt/Training-Module/issues/362
https://github.com/buerokratt/Training-Module/issues/400

### CODE REVIEW

File: DSL/Resql/training/get-available-intents.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
+WITH connected_intents AS
```
Comment (1AhmedYasser) [inline DSL/Resql/training/get-available-intents.sql:1]: Please replace this query with this enhanced one to cover more edge cases

```
WITH connected_intents AS
  (SELECT intent,
          service,
          service_name,
          status,
          created
   FROM service_trigger
   WHERE (intent,
          service,
          service_name,
          created) IN
       (SELECT intent,
               service,
               service_name,
               max(created)
        FROM service_trigger
        GROUP BY intent,
                 service,
                 service_name)
     AND status in ('pending',
                    'approved'))
SELECT intent
FROM intent
WHERE intent NOT IN
    (SELECT intent
     FROM connected_intents)
ORDER BY intent ASC
```

File: GUI/src/pages/Training/Intents/IntentExamplesEntry.tsx [hunk 2]
```diff
@@ -54,44 +58,66 @@ const IntentExamplesEntry: FC<IntentExamplesEntryProps> = ({ value, entities, on
     console.log(data);
```
Comment (1AhmedYasser) [inline GUI/src/pages/Training/Intents/IntentExamplesEntry.tsx:58]: Please remove this console.log

**Review by 1AhmedYasser — CHANGES_REQUESTED** [review]

**Comment by 1AhmedYasser** [comment]
@bartekpospiech Question: At which screen or modal do you send a trigger request ?

File: GUI/src/pages/Training/AutoServices/index.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        id: 'space',
+        meta: {
+          size: '1%',
+        },
+      }),
+    ],
+    [appRequestColumnHelper, t]
+  );
+
+  console.log(intents, triggers);
```
Comment (1AhmedYasser) [inline GUI/src/pages/Training/AutoServices/index.tsx:120]: Please remove console.log

**Review by 1AhmedYasser — CHANGES_REQUESTED** [review]

**Comment by 1AhmedYasser** [comment]
@bartekpospiech Please resolve conflicts

**Comment by 1AhmedYasser** [comment]
> _ No description provided. _

Also please add a description and link the task to it, similar to what i did here https://github.com/buerokratt/Service-Module/pull/214

File: DSL/Resql/training/get-requested-service-triggers.sql [hunk 4]
```diff
@@ -0,0 +1,10 @@
+SELECT intent,
+        service,
+        MAX(service_name) AS service_name,
+        MAX(created) AS requested_at,
+        MAX(author_role) as author_role 
+ FROM service_trigger
+ GROUP BY intent,
+          service
+ HAVING MAX(status) = 'pending'
+ AND max("author_role") != 'service_manager';
```
Comment (1AhmedYasser) [inline DSL/Resql/training/get-requested-service-triggers.sql:10]: modify this to `AND max("author_role") != 'trainer';`

**Review by 1AhmedYasser — CHANGES_REQUESTED** [review]

**Comment by bartekpospiech** [comment]
> @bartekpospiech Question: At which screen or modal do you send a trigger request ? where is `setServiceTrigger` called in pages or modal

it's done in modal at the bottom of `GUI/src/pages/Training/Intents/index.tsx` file

**Comment by 1AhmedYasser** [comment]
> > @bartekpospiech Question: At which screen or modal do you send a trigger request ? where is `setServiceTrigger` called in pages or modal
> 
> it's done in modal at the bottom of `GUI/src/pages/Training/Intents/index.tsx` file

Ok i checked it, i think you should call the request permission here for what i understood from the code

```
 <Button appearance="secondary" onClick={onModalClose}>
                Tühista
              </Button>
              <Button
                onClick={() =>
                  useIntentsListStore
                    .getState()
                    .setSelectedService(connectedService as unknown as Service)
                }
              >
                Jah
              </Button>
```

**Pull request closed without merging**